pandas/io/parsers.py

```diff
@@ -997,14 +997,14 @@ def _validate_usecols_arg(usecols):
 
     if usecols is not None:
         if callable(usecols):
-            return usecols
+            return usecols, None
         usecols_dtype = lib.infer_dtype(usecols)
         if usecols_dtype not in ('empty', 'integer',
                                  'string', 'unicode'):
             raise ValueError(msg)
 
-        return set(usecols)
-    return usecols
+        return set(usecols), usecols_dtype
```

Comment:
still, why are we returning a ``set`` here? why not a list? (for this same case)? then you don't have to rely on the ordering and convert to a list if its integer type? 

could just do this: http://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-in-whilst-preserving-order

then make it into a set at the last moment.

I am nitpicking on this because converting this to a set is happening too early, and it is just confusing.